homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,268 @@
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_unit
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self.device.indoorTemperature
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the hvac mode."""
+        return HVAC_MODES[self.device.changeableValues.mode]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """List of available hvac modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        device: LyricDevice = self.device
+        if not device.hasDualSetpointStatus:
+            return device.changeableValues.heatSetpoint
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.device
+        if device.hasDualSetpointStatus:
+            return device.changeableValues.coolSetpoint
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.device
+        if device.hasDualSetpointStatus:
+            return device.changeableValues.heatSetpoint
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return current preset mode."""
+        return self.device.changeableValues.thermostatSetpointStatus
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return preset modes."""
+        return [
+            PRESET_NO_HOLD,
+            PRESET_HOLD_UNTIL,
+            PRESET_PERMANENT_HOLD,
+            PRESET_TEMPORARY_HOLD,
+            PRESET_VACATION_HOLD,
+        ]
+
+    @property
+    def min_temp(self) -> float:
+        """Identify min_temp in Lyric API or defaults if not available."""
+        device: LyricDevice = self.device
+        return (
+            device.minCoolSetpoint
+            if LYRIC_HVAC_MODE_COOL in device.allowedModes
+            else device.minHeatSetpoint
+        )
+
+    @property
+    def max_temp(self) -> float:
+        """Identify max_temp in Lyric API or defaults if not available."""
+        device: LyricDevice = self.device
+        return (
+            device.maxHeatSetpoint
+            if LYRIC_HVAC_MODE_HEAT in device.allowedModes
+            else device.maxCoolSetpoint
+        )
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+
+        device: LyricDevice = self.device
+        if device.hasDualSetpointStatus:
+            if target_temp_low is not None and target_temp_high is not None:
```

Comment:
`temp` will be unbound if this `if` is false

Reply:
Please invert this check and raise the error if true. Then we can outdent below.